### PR TITLE
fix the bug: if LITTLEFS_SDMMC_SUPPORT turns on, some features are pr…

### PR DESCRIPTION
**Pull request by wang33winner**
…oblematic when using flash.
some macro for CONFIG_LITTLEFS_SDMMC_SUPPORT will drop the work  for flash

### CODE REVIEW

File: src/esp_littlefs.c [hunk 1]
```diff
@@ -253,13 +253,12 @@ esp_err_t format_from_efs(esp_littlefs_t *efs)
         if (efs->sdcard) {
             res = lfs_format(efs->fs, &efs->cfg);
         } else
-#else
+#endif
```
Comment (BrianPugh) [inline src/esp_littlefs.c:256]: real change

File: src/esp_littlefs.c [hunk 2]
```diff
@@ -1072,11 +1071,10 @@ static esp_err_t esp_littlefs_init(const esp_vfs_littlefs_conf_t* conf)
             if (conf->sdcard) {
                 err = esp_littlefs_format_sdmmc(conf->sdcard);
             } else
-#else
+#endif
```
Comment (BrianPugh) [inline src/esp_littlefs.c:1074]: real change

File: src/esp_littlefs.c [hunk 3]
```diff
@@ -1093,15 +1091,14 @@ static esp_err_t esp_littlefs_init(const esp_vfs_littlefs_conf_t* conf)
         efs->cache = esp_littlefs_calloc(sizeof(*efs->cache), efs->cache_size);
 
         if(conf->grow_on_mount){
-#ifndef CONFIG_LITTLEFS_SDMMC_SUPPORT
-            res = lfs_fs_grow(efs->fs, efs->partition->size / efs->cfg.block_size);
-#else
+#ifdef CONFIG_LITTLEFS_SDMMC_SUPPORT
```
Comment (BrianPugh) [inline src/esp_littlefs.c:1094]: real change

**Review by BrianPugh — COMMENTED** [review]
@huming2207 can you take a look at this? There appears to be 3 real changes among the formatting/linting changes.

File: src/esp_littlefs.c [hunk 4]
```diff
@@ -1093,15 +1091,14 @@ static esp_err_t esp_littlefs_init(const esp_vfs_littlefs_conf_t* conf)
         efs->cache = esp_littlefs_calloc(sizeof(*efs->cache), efs->cache_size);
 
         if(conf->grow_on_mount){
-#ifndef CONFIG_LITTLEFS_SDMMC_SUPPORT
-            res = lfs_fs_grow(efs->fs, efs->partition->size / efs->cfg.block_size);
-#else
```
Comment (wang33winner) [inline src/esp_littlefs.c:1098]: need as well

File: src/esp_littlefs.c [hunk 5]
```diff
@@ -1093,15 +1091,14 @@ static esp_err_t esp_littlefs_init(const esp_vfs_littlefs_conf_t* conf)
...
+#ifdef CONFIG_LITTLEFS_SDMMC_SUPPORT
             if (efs->sdcard) {
                 res = lfs_fs_grow(efs->fs, efs->sdcard->csd.capacity);
-            } else {
```
Comment (wang33winner) [inline src/esp_littlefs.c:1101]: need as well

File: src/esp_littlefs.c [hunk 6]
```diff
@@ -1093,15 +1091,14 @@ static esp_err_t esp_littlefs_init(const esp_vfs_littlefs_conf_t* conf)
...
+            } else
+#endif
+            {
                 res = lfs_fs_grow(efs->fs, efs->partition->size / efs->cfg.block_size);
             }
-#endif
```
Comment (wang33winner) [inline src/esp_littlefs.c:1104]: need as well

File: src/esp_littlefs.c [hunk 7]
```diff
@@ -253,13 +253,12 @@ esp_err_t format_from_efs(esp_littlefs_t *efs)
...
             res = lfs_format(efs->fs, &efs->cfg);
         } else
-#else
+#endif
         {
             efs->cfg.block_count = efs->partition->size / efs->cfg.block_size;
             res = lfs_format(efs->fs, &efs->cfg);
             efs->cfg.block_count = 0;
         }
-#endif
```
Comment (wang33winner) [inline src/esp_littlefs.c:262]: need as well

File: src/esp_littlefs.c [hunk 8]
```diff
@@ -1072,11 +1071,10 @@ static esp_err_t esp_littlefs_init(const esp_vfs_littlefs_conf_t* conf)
             if (conf->sdcard) {
                 err = esp_littlefs_format_sdmmc(conf->sdcard);
             } else
-#else
+#endif
             {
                 err = esp_littlefs_format_partition(efs->partition);
             }
-#endif
```
Comment (wang33winner) [inline src/esp_littlefs.c:1079]: need as well

**Comment by wang33winner** [comment]
> @huming2207 can you take a look at this? There appears to be 3 real changes among the formatting/linting changes.

yes, some "endif" modifications are also required, others are formatting/linting changes.

**Comment by BrianPugh** [comment]
the code changes look good to me, I'll merge in a day or two (unless huming2207 provides back sooner).

**Review by BrianPugh — APPROVED** [review]

**Comment by huming2207** [comment]
I've reviewed the code - frankly speaking I don't know what is this for. Instead, this PR will definitely break SD card support.
 
I really don't think the `LITTLEFS_SDMMC_SUPPORT` will affect any existing LittleFS functionalities on SPI NOR flash over `esp_partition`. My colleagues are also using [v1.13](https://github.com/joltwallet/esp_littlefs/releases/tag/v1.13.0) for production and they have no complaints so far. I will double check with them on next Monday.

**Comment by huming2207** [comment]
Oh I'm really sorry, I misread the code on iPad 😅😅 

I think it's fine. I will test it with our real devices and see if that actually cause any issues on Monday. On the other hand so far I didn't hear any report about breaking on `esp_partition` on SPI flash though, I will double check again with my colleagues...

**Comment by huming2207** [comment]
Hi @BrianPugh we've tested it and we believe it's all good to go.

**Pull request merged**